Convert the services blog and work pages to streamfield blocks, and style them

helenb opened the pull request:
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1371208170) and [also this ticket](https://torchbox.monday.com/boards/1192293412/pulses/1371207967)

### Description of Changes Made

- Converts the work page chooser and blog page chooser to streamfield blocks - they were previously fields
- Styling for the blog page chooser and work page chooser
- Slightly refactors the spacing above related posts
- Re-instates the `work_index` property as we can't use a relative link on the tag links on a service page - and makes this and the blog one cached so they don't run multiple times on a request.
- Moves the reading time and date into an include to avoid repeated code in lots of places

### How to Test

### Screenshots

<details>
  <summary>Expand to see more</summary>

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [x] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [x] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

helenb commented on tbx/project_styleguide/templates/patterns/atoms/date_and_reading_time/date_and_reading_time.html:
```diff
@@ -0,0 +1,9 @@
+{% if date %}
```
Refactors a common pattern into a component

helenb commented on tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:
```diff
@@ -59,7 +54,7 @@ <h1 class="heading heading--one-b">{{ page.title }}</h1>
             <ul class="grid__related-posts streamfield__related-posts">
                 {% for post in page.related_blog_posts %}
-                    {% include "patterns/molecules/listing/listing--avatar.html" with title=post.title name=post.author.name job_title=post.author.role link=post.url date=post.date reading_time=post.read_time tags=post.related_services listing_image=post.listing_image avatar=post.author.image %}
+                    {% include "patterns/molecules/listing/listing--avatar.html" with title=post.title name=post.author.name job_title=post.author.role link=post.url date=post.date reading_time=post.read_time tags=post.related_services avatar=post.author.image %}
```
removes an unused attribute

helenb commented on tbx/work/models.py:
```diff
@@ -269,6 +273,21 @@ def read_time(self):
+        return self.header_image
+
+    @property
+    def work_index(self):
```
I had to add this back, as we can't use a relative link for the tag link base on service pages.

engineervix commented on tbx/services/migrations/0007_alter_servicepage_body.py:
This migration and the previous one (0006) can be in one file.

**Suggesttion**: Delete `s/0006_move_blog_and_work_chooser_to_streamfields.py` and `0007_alter_servicepage_body.py` then re-create the migration

helenb replied:
Thanks @engineervix - is that just to avoid making deployments etc too slow with lots of migrations? I wasn't sure if it was better practice to separate out unrelated migrations.

helenb replied:
Updated in ac7f3f071ba53d44c8a33680328694238949c3ee

engineervix commented on tbx/work/models.py:
```diff
@@ -134,6 +134,10 @@ def read_time(self):
             return "x"
 
+    @property
+    def listing_image(self):
```
:+1:

engineervix commented on tbx/work/models.py:
```diff
@@ -268,6 +272,21 @@ def read_time(self):
             return "x"
 
+    @property
+    def listing_image(self):
```
:+1:

engineervix replied:
With the addition of this property on both Work Pages and Historical Work Pages, then it would be a good idea to update this accordingly

https://github.com/torchbox/torchbox.com/blob/691e7cb3b35092637cd3325aee46b002be3daad5/tbx/work/models.py#L353-L354

So that we now have

```python
"listing_image": work.listing_image,
```

helenb replied:
Updated in 9c96331240565cc7ba5c9ce103467f6a9311bd9f

engineervix reviewed (commented):
Thanks @helenb 

Left a few suggestions

chris-lawton commented on tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:
```diff
@@ -44,22 +44,17 @@
 
         {% if page.related_blog_posts %}
-            {% include "patterns/atoms/motif-heading/motif-heading.html" with heading="More" heading_level=2 classes="section-title--related-posts motif-heading--two" aria_label="Related blog posts" %}
+            {% include "patterns/atoms/section-title/section-title.html" with title="More" aria_label="Related blog posts" classes="section-title--related-posts section-title--large-gap" %}
```
Thanks!

chris-lawton commented on tbx/project_styleguide/templates/patterns/atoms/date_and_reading_time/date_and_reading_time.html:
```diff
@@ -0,0 +1,9 @@
+{% if date %}
+    <time datetime="{{ date|date:'Y-m-d' }}">{{ date }}</time>
```
This might need updating based on the changes in https://github.com/torchbox/torchbox.com/pull/83/files

helenb replied:
Yes I'll rebase if necessary, but there are some questions over that MR at the moment

chris-lawton commented on tbx/static_src/sass/components/_work-hero.scss:
```diff
@@ -49,8 +49,4 @@
-
-    &__divider {
-        margin: 0 7px;
-    }
```
Nice to delete all of these!

chris-lawton commented:
Thanks @helenb - this looks great. Should `blog_chooser_block.html` or `work_chooser_block.html` be added to the blog/work page template in the pattern lib and/or the component list?

helenb commented:
> Thanks @helenb - this looks great. Should `blog_chooser_block.html` or `work_chooser_block.html` be added to the blog/work page template in the pattern lib and/or the component list?

I didn't add it to the stylguide as it felt like similar patterns to what we had before, but I've added them to the service page, which is where they will appear (and fixed the featrued case study image yaml for that page)
